src/mlpack/bindings/python/mlpack/matrix_utils.py

```diff
@@ -54,19 +54,26 @@ def to_matrix(x, dtype=np.double, copy=False):
       return x, False
   elif (isinstance(x, np.ndarray) and x.dtype == dtype and x.flags.f_contiguous):
     if copy: # Copy the matrix if required.
-      return np.ndarray(x.shape, buffer=x.flatten(), dtype=dtype, order='C').copy("C"), True
+      return np.ndarray(x.shape, buffer=x.flatten(), dtype=dtype,
+          order='C').copy("C"), True
     else:
-      return np.ndarray(x.shape, buffer=x.flatten(), dtype=dtype, order='C'), False
+      return np.ndarray(x.shape, buffer=x.flatten(), dtype=dtype, order='C'), \
+          False
   else:
     if isinstance(x, pd.core.series.Series) or isinstance(x, pd.DataFrame):
+      # We can only avoid a copy if the dtype is the same and the copy flag is
+      # false.  I'm actually not sure if this is possible, since in everything I
+      # have found, Pandas stores with F_CONTIGUOUS not C_CONTIGUOUS.
       y = x.values
-      if copy: # Copy the matrix if required.
-        return np.ndarray(y.shape, buffer=y.flatten(), dtype=dtype, order='C').copy("C"), True
+      if copy == False and y.dtype == dtype and y.flags.c_contiguous:
```

Comment:
@rcurtin Did you get a chance to run this code and see if that's the case?

Reply:
Hey, sorry for the slow response on this one.  It has not been a good two weeks.  @Yashwants19 I wasn't able to find any documentation clarifying whether Pandas uses `F_CONTIGUOUS` or `C_CONTIGUOUS` only.  I believe that you are right that `F_CONTIGUOUS` is generally used, and any Pandas matrix I found or created seemed to be `F_CONTIGUOUS`.  However unfortunately if the matrix is `F_CONTIGUOUS` we must *always* transpose (thus copy) since observations are rows (in Python) and we need it to be row-major (or `C_CONTIGUOUS`) to pass correctly to mlpack (where it will be column-major).

Reply:
@Yashwants19 let me know if that's a decent explanation---if you think I got something wrong, I definitely could have.  So I'll wait until you respond to merge (or otherwise I'll just merge in a few days). :+1: